### ISSUE TITLE
Why not pad embedding_kernel when use fp32 in gpt?

### ISSUE THREAD

**Issue by zbtong** [NVIDIA/FasterTransformer#165]
Hello, I found when calculate the logits after encoders, the `n` argument passed to method `cublasMM_cublasLtMM_wrapper_decoder`

```c++
DataType_ alpha = DataType_(1.0f);
DataType_ beta = DataType_(0.0f);
assert(args_.vocab_size_padded_ % t_parallel_param_.world_size == 0);
int n = args_.vocab_size_padded_ / t_parallel_param_.world_size;

if(t_parallel_param_.world_size == 1)
{
    PUSH_RANGE("After Transformer/GEMM")
    cublasMM_cublasLtMM_wrapper_decoder(decoding_params.cublaslt_handle, 
                                        decoding_params.cublas_handle, 
                                        CUBLAS_OP_T, CUBLAS_OP_N,
                                        n, local_batch, k,
                                        &alpha,
                                        embedding_kernel_ptr, AType_, k,
                                        decoder_normed_result_buf_, BType_, k,
                                        &beta,
                                        logits_buf_, CType_, n,
                                        decoding_params.stream, cublasAlgoMap_,
                                        cublas_workspace_);
    POP_RANGE
}
```

but when use fp32, the embedding_kernel_ptr is pointed to decoding_params.embedding_table, whose shape is [vocab_size, global_hidden_units], `vocab_size` always smaller than `vocab_size_padd`。As following code show:

```c++
args_.vocab_size_padded_ = div_up(args_.vocab_size_, 64) * 64;
```

```c++
if(std::is_same<DataType_, float>::value || (std::is_same<DataType_, half>::value && args_.vocab_size_padded_ == args_.vocab_size_))
{
    embedding_kernel_ptr = (const DataType_ *)decoding_params.embedding_kernel;
}
else
{
    cudaMemcpyAsync(embedding_kernel_padded_, decoding_params.embedding_kernel, 
                    sizeof(DataType_) * args_.vocab_size_ * args_.hidden_units_, cudaMemcpyDeviceToDevice, decoding_params.stream);
    embedding_kernel_ptr = (const DataType_ *)embedding_kernel_padded_;
}
```

Could this cause some runtime error?

**Comment by byshiue** [comment]
It does not cause runtime error. When using FP16, tensor core can bring better perf with padding.

**Comment by byshiue** [comment]
Close this bug because it is inactivated. Feel free to re-open this issue if you still have any problem.
